src/python/gudhi/representations/vector_methods.py

```diff
@@ -235,6 +205,7 @@ def __init__(self, weight=lambda x: 1, resolution=100, sample_range=[np.nan, np.
             sample_range ([double, double]): minimum and maximum for the weighted average domain, of the form [x_min, x_max] (default [numpy.nan, numpy.nan]). It is the interval on which samples will be drawn evenly. If one of the values is numpy.nan, it can be computed from the persistence diagrams with the fit() method.
         """
         self.weight, self.resolution, self.sample_range = weight, resolution, sample_range
+        self.im_range = None
```

Comment:
Is that useful?

Reply:
No, my bad (editor refactoring suggestions)